signalfx-java/src/main/java/com/signalfx/signalflow/ServerSentEventsTransport.java

```diff
@@ -0,0 +1,531 @@
+        try {
+            if (response != null) {
+                response.close();
+            }
+        } catch (IOException ioex) {
+            log.error("error closing response", ioex);
+        }
+    }
+
+    private void close(TransportConnection connection) {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+        } catch (IOException ioex) {
+            log.error("error closing transport connection", ioex);
+        }
+    }
+
+    /**
+     * Builder of SSE Transport Instance
+     */
+    public static class TransportBuilder {
+
+        private String token;
+        private String protocol = "https";
+        private String host = "stream.signalfx.com";
+        private int port = 443;
+        private int timeout = 1;
+        private int version = 2;
+
+        public TransportBuilder(String token) {
+            this.token = token;
+        }
+
+        public TransportBuilder setProtocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public TransportBuilder setHost(String host) {
+            this.host = host;
+            return this;
+        }
+
+        public TransportBuilder setPort(int port) {
+            this.port = port;
+            return this;
+        }
+
+        public TransportBuilder setTimeout(int timeout) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public TransportBuilder setAPIVersion(int version) {
+            this.version = version;
+            return this;
+        }
+
+        public ServerSentEventsTransport build() {
+            SignalFxEndpoint endpoint = new SignalFxEndpoint(this.protocol, this.host, this.port);
+            ServerSentEventsTransport transport = new ServerSentEventsTransport(this.token,
+                    endpoint, this.version, this.timeout * 1000);
+            return transport;
+        }
+    }
+
+    /**
+     * SSE Transport Connection
+     */
+    public static class TransportConnection extends AbstractHttpReceiverConnection {
+
+        protected static final Logger log = LoggerFactory.getLogger(TransportConnection.class);
+        public static final int DEFAULT_TIMEOUT_MS = 1000;
+        protected final RequestConfig transportRequestConfig;
+
+        public TransportConnection(SignalFxEndpoint endpoint) {
+            this(endpoint, DEFAULT_TIMEOUT_MS);
+        }
+
+        public TransportConnection(SignalFxEndpoint endpoint, int timeoutMs) {
+            super(endpoint, timeoutMs, new BasicHttpClientConnectionManager());
+
+            this.transportRequestConfig = RequestConfig.custom().setSocketTimeout(0)
+                    .setConnectionRequestTimeout(this.requestConfig.getConnectionRequestTimeout())
+                    .setConnectTimeout(this.requestConfig.getConnectTimeout())
+                    .setProxy(this.requestConfig.getProxy()).build();
+
+            if (log.isDebugEnabled()) {
```

Comment:
You don't really need to gate your logging statements with this check (it's the first thing `log.debug()` will do), unless the methods you're calling on your parameters are expensive and you're logging in a hot path. It doesn't seem to to be the case here (and most everywhere else) so you can drop those.